lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -88,6 +99,9 @@ impl ClockMap {
     /// Returns whether the clock was updated (or initialized) and the current tick.
     #[must_use = "clock update status and current tick must be used"]
     fn advance_clock_impl(&mut self, clock_tag: ClockTag) -> (bool, u64) {
+        // To keep things simple, assume we always change state here
+        self.changed = true;
```

Comment:
We can consider being smarter about the changed state here. Currently I just assume we change the state in all cases.

I didn't want to make things more complicated by passing 'changed' booleans all over the place to make this work.

Thoughts/opionions?

Reply:
Our documentation is a bit off here.

It returns whether it was accepted, not whether it is updated. Those are not the same. See:

https://github.com/qdrant/qdrant/blob/c2aa229e59c58c5d690b9dacb97e72d8dd7d27b2/lib/collection/src/shards/local_shard/clock_map.rs#L173-L182

Shall we just use this accepted flag then to distinguish between changed or not? If we do that, we still set the changed flag on duplicate clock tags that don't actually change any state. But I guess that's a good enough middle ground.

Reply:
I've implemented this now. We only set `changed = true` if some clock tag is _accepted_.

Reply:
Yeah, I think it's still a bit better. Some false positives, but a bit less than with just `changed = true` always.

Reply:
By the way, I've updated documentation around the functions accordingly. They now mention that it returns whether the clock tag was accepted.